jdaviz/configs/imviz/plugins/aper_phot_simple/aper_phot_simple.py

```diff
@@ -132,31 +138,63 @@ def _subset_selected_changed(self, event={}):
         try:
             self._selected_subset = self._get_region_from_subset(subset_selected)
             self._selected_subset.meta['label'] = subset_selected
+
+            if isinstance(self._selected_subset, CirclePixelRegion):
+                self.bg_annulus_inner_r = self._selected_subset.radius
+            elif isinstance(self._selected_subset, EllipsePixelRegion):
+                self.bg_annulus_inner_r = max(self._selected_subset.width,
+                                              self._selected_subset.height) * 0.5
+            elif isinstance(self._selected_subset, RectanglePixelRegion):
+                self.bg_annulus_inner_r = np.sqrt(self._selected_subset.width ** 2 +
+                                                  self._selected_subset.height ** 2) * 0.5
+            else:  # pragma: no cover
+                raise TypeError(f'Unsupported region shape: {self._selected_subset.__class__}')

```

Comment:
I suppose an error is helpful here so that we add support for it, but does this then prevent manually entering the annulus radius/width (especially since the exception resets the subset... although that might leave the UI in an unrecoverable state)? Maybe this is a case for a snackbar and then just default to an inner radius of some fixed number.

Reply:
> does this then prevent manually entering the annulus radius/width

Actually, I am not sure. I have not tested with selecting Aperture from a weird Subset that is like a result of subtracting/etc other Subsets (because mainly I don't know how to do it properly and I don't see a real science case to select it for aperture photometry).

Reply:
If you manually load weird subsets via the Astrowidgets API, the logic I put in should make those "invisible" to the plugin in the first place, so those are not a problem. (Unless somehow my logic was overwritten since.)

Reply:
Update: So, even after loading weird shapes using Astrowidgets API, the plugin is "smart" enough to exclude them from the dropdown (because of the "Subset" prefix check) but this might accidentally break when we implement the feature to allow users to rename subset, so I still prefer to leave this logic in.

> does this then prevent manually entering the annulus radius/width (especially since the exception resets the subset... although that might leave the UI in an unrecoverable state)?

I am not sure if this is possible currently. The `else` is just there for completeness right now. If we ever see an exception here, it is a serious bug and we need to add new code to prevent it properly. In that case, I think it is clearer to see ugly traceback because Snackbar can be easily dismissed.